src/main/java/seedu/command/DeleteCommand.java

```diff
@@ -32,15 +36,23 @@ public DeleteCommand(ArrayList<String> commandStrings) {
      */
     public CommandResult execute() {
         String equipmentName;
+        String serialNumber;
+
         try {
-            equipmentName = equipmentManager.getEquipmentList().get(commandStrings.get(0)).getItemName();
+            serialNumber = prepareDelete();
+            if (serialNumber.equals("")) {
+                return new CommandResult(ONLY_SN_ACCEPTED);
+            }
```

Comment:
You might need to adjust this based on the logic in the other comment further down.